pkg/database/realm.go

```diff
@@ -1159,3 +1160,38 @@ func ToCIDRList(s string) ([]string, error) {
 	sort.Strings(cidrs)
 	return cidrs, nil
 }
+
+// RealmUserStats carries the per-user-per-day-per-realm Codes issued.
+// This is a structure joined from multiple tables in the DB.
+type RealmUserStats struct {
+	UserID      uint
+	Name        string
+	CodesIssued uint
+	Date        time.Time
+}
+
+// CodesPerUser returns a set of UserStats for a given date range.
+func (r *Realm) CodesPerUser(db *Database, start, stop time.Time) ([]*RealmUserStats, error) {
```

Comment:
I think there's a subtle issue here w/ removing a user from a realm.

we likely need to add tracking to "soft-delete" a user from a realm until they roll off of the stats.